go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/rancher/rke2
 
-go 1.19
+go 1.21
+
+toolchain go1.22.0
```

Comment:
golang REALLY wants to do this now, but don't let it. If your system golang is 1.22+ you now need export GOTOOLCHAIN in order to avoid having golang mangle your go.mod:

`GOTOOLCHAIN=go1.20.13 go mod tidy`

where `go1.20.13` is whatever version of golang we're going to use at build time.